File: CRPL.Web/ClientApp/src/app/_Models/Applications/DisputeViewModel.ts
import {ApplicationViewModel} from "./ApplicationViewModel";
import {DisputeType} from "./DisputeInputModel";
import {RegisteredWorkViewModel} from "../Works/RegisteredWork";
import {UserAccountMinimalViewModel} from "../Account/UserAccountMinimalViewModel";

export enum ExpectedRecourse
{
  ChangeOfOwnership,
  Payment
}

export enum ResolvedStatus
{
  Created,
  NeedsOnChainAction,
  Processing,
  Resolved,
  Failed
}

export interface ResolveResult
{
  Rejected: boolean;
  ResolvedStatus: ResolvedStatus;
  Transaction?: string;
  TransactionUri?: string;
  Message?: string;
}

export interface DisputeViewModel extends ApplicationViewModel
{
  DisputeType: DisputeType;
  Reason:string;
  Spotted: Date;
  Infractions: number;
  ExpectedRecourse: ExpectedRecourse;
  ExpectedRecourseData: string;
  ContactAddress: string;
  LinkToInfraction: string;
  ResolveResult: ResolveResult;

  DisputedWork: RegisteredWorkViewModel;
  Accuser: UserAccountMinimalViewModel;
}

File: CRPL.Data/Applications/ViewModels/DisputeViewModel.cs
using CRPL.Data.Account;
using CRPL.Data.Account.ViewModels;
using CRPL.Data.Applications.Core;

namespace CRPL.Data.Applications.ViewModels;

public class DisputeViewModel : ApplicationViewModel
{
    public DisputeType DisputeType { get; set; }
    public string Reason { get; set; }
    public DateTime? Spotted { get; set; }
    public int? Infractions { get; set; }
    public ExpectedRecourse? ExpectedRecourse { get; set; }
    public string? ExpectedRecourseData { get; set; }
    public string? ContactAddress { get; set; }
    public string? LinkToInfraction { get; set; }
    public ResolveResultWithUri ResolveResult { get; set; }

    public RegisteredWorkViewModel DisputedWork { get; set; }
    public UserAccountMinimalViewModel Accuser { get; set; }
}

public class DisputeViewModelWithoutAssociated : ApplicationViewModelWithoutAssociated
{
    
}